nutils/evaluable.py

```diff
@@ -1065,6 +1065,41 @@ def _intbounds(self):
   def _intbounds_impl(self):
     return float('-inf'), float('inf')
 
+class SimpleArrayHelper(Array):
+
+  def __init__(self, simple_array, lower, points_shape, *args):
+    self.simple_array = simple_array
+    self.lower = lower
+    self.points_shape = points_shape
+    self.args = args
+    super().__init__((points_shape, *args), shape=points_shape+self.simple_array.shape, dtype=simple_array.dtype)
+
+  def evalf(self, points_shape, *args):
+    points_shape = tuple(map(int, points_shape))
+    points_dim = len(points_shape)
+    npoints = util.product(points_shape, 1)
+    result = self.simple_array.evalf(*(arg.reshape(npoints, *arg.shape[points_dim:]) for arg in args))
+    assert result.ndim == self.ndim + 1 - points_dim
+    if not args and result.shape[0] == 1 and npoints != 1:
+      return numpy.broadcast_to(result, points_shape + result.shape[1:])
+    else:
+      return result.reshape(points_shape + result.shape[1:])
+
+  def _derivative(self, var, seen):
+    if self.dtype != float:
+      return super()._derivative(var, seen)
+    result = Zeros(self.shape + var.shape, dtype=self.dtype)
+    points_dim = len(self.points_shape)
+    for iarg, arg in enumerate(self.args):
+      if arg.dtype != float or var not in arg.dependencies and var != arg:
```

Comment:
So `SimpleArray` does not need to define these cases, something to add to the docstring?

Reply:
Only the case where an argument is not a `float`. That an argument does not depend on `var` cannot be predicted. I'll add this to the docs.